### ISSUE TITLE
"invalid UTF8 characters" error when parsing pdf

### ISSUE THREAD

**Issue by eazy2k13** [wallabag/wallabag#3576]
### Issue details

A link pointing to an pdf document should be grabbed by wallabag:
https://www.swm.de/dam/swm/pressemitteilungen/2018/01/20180124-swm-mvg-tum-e-trike-fuer-mvg-rad.pdf
After adding this link (or the bit.ly shortlink ) to wallabag the application is not usable anymore.
The android app does not sync anymore and visiting the desktop site shows the following error:
"500: Internal Server Error - The string to escape is not a valid UTF-8 string"

The f43.me test (internal) can parse this link/document.

It seems related to issue #3154 

```
[2018-01-24 15:55:48] graby.DEBUG: Graby is ready to fetch [] []
[2018-01-24 15:55:48] graby.DEBUG: . looking for site config for bit.ly in primary folder {"host":"bit.ly"} []
[2018-01-24 15:55:48] graby.DEBUG: Appending site config settings from global.txt [] []
[2018-01-24 15:55:48] graby.DEBUG: . looking for site config for global in primary folder {"host":"global"} []
[2018-01-24 15:55:48] graby.DEBUG: ... found site config global.txt {"host":"global.txt"} []
[2018-01-24 15:55:48] graby.DEBUG: Cached site config with key: bit.ly {"key":"bit.ly"} []
[2018-01-24 15:55:48] graby.DEBUG: . looking for site config for global in primary folder {"host":"global"} []
[2018-01-24 15:55:48] graby.DEBUG: ... found site config global.txt {"host":"global.txt"} []
[2018-01-24 15:55:48] graby.DEBUG: Appending site config settings from global.txt [] []
[2018-01-24 15:55:48] graby.DEBUG: Cached site config with key: global {"key":"global"} []
[2018-01-24 15:55:48] graby.DEBUG: Cached site config with key: bit.ly.merged {"key":"bit.ly.merged"} []
[2018-01-24 15:55:48] graby.DEBUG: Fetching url: http://bit.ly/2G7LWPg {"url":"http://bit.ly/2G7LWPg"} []
[2018-01-24 15:55:48] graby.DEBUG: Trying using method "get" on url "http://bit.ly/2G7LWPg" {"method":"get","url":"http://bit.ly/2G7LWPg"} []
[2018-01-24 15:55:48] graby.DEBUG: Use default user-agent "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/535.2 (KHTML, like Gecko) Chrome/15.0.874.92 Safari/535.2" for url "http://bit.ly/2G7LWPg" {"user-agent":"Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/535.2 (KHTML, like Gecko) Chrome/15.0.874.92 Safari/535.2","url":"http://bit.ly/2G7LWPg"} []
[2018-01-24 15:55:48] graby.DEBUG: Use default referer "http://www.google.co.uk/url?sa=t&source=web&cd=1" for url "http://bit.ly/2G7LWPg" {"referer":"http://www.google.co.uk/url?sa=t&source=web&cd=1","url":"http://bit.ly/2G7LWPg"} []
[2018-01-24 15:55:48] graby.DEBUG: Data fetched: [array] {"data":{"effective_url":"https://www.swm.de/dam/swm/pressemitteilungen/2018/01/20180124-swm-mvg-tum-e-trike-fuer-mvg-rad.pdf","body":"(only length for debug): 139113","headers":"application/pdf;charset=UTF-8","all_headers":{"date":"Wed, 24 Jan 2018 14:55:48 GMT","x-frame-options":"SAMEORIGIN","cache-control":"max-age=600, public","expires":"Wed, 24 Jan 2018 15:05:48 GMT","last-modified":"Wed, 24 Jan 2018 08:17:47 GMT","content-length":"139113","content-type":"application/pdf;charset=UTF-8","server":"SWM Webserver","set-cookie":"NSC_MC_txn=5ccba3d89bf6de808cd2d2d97dd23179080c110890a878b69ef5789d405983812c30da00;path=/;secure;httponly"},"status":200}} [] 
[2018-01-24 15:55:48] graby.DEBUG: Filtering HTML to remove XSS [] []
[2018-01-24 15:55:48] app.DEBUG: DownloadImages: Folder used for that Entry id {"folder":"/path/to/wallabag/web/assets/images/0/a/0acff686","entryId":3100} []
[2018-01-24 15:55:48] app.DEBUG: DownloadImagesSubscriber: updated html. [] []
[2018-01-24 15:55:48] app.DEBUG: DownloadImages: Folder used for that Entry id {"folder":"/path/to/wallabag/web/assets/images/0/a/0acff686","entryId":3100} []
[2018-01-24 15:55:48] app.DEBUG: DownloadImages: working on image:  [] []
[2018-01-24 15:55:48] app.DEBUG: DownloadImages: Checking extension {"ext":null,"header":"application/pdf;charset=UTF-8"} []
[2018-01-24 15:55:48] app.DEBUG: DownloadImages: Checking extension (alternative) {"ext":null} []
[2018-01-24 15:55:48] app.ERROR: DownloadImages: Processed image with not allowed extension. Skipping:  [] []
[2018-01-24 15:55:48] app.ERROR: DownloadImages: Error while regenerating image {"path":"/path/to/wallabag/web/assets/images/0/a/0acff686/f5f201ac."} []
[2018-01-24 15:55:48] request.CRITICAL: Uncaught PHP Exception RuntimeException: "Your data could not be encoded because it contains invalid UTF8 characters." at /path/to/wallabag/vendor/jms/serializer/src/JMS/Serializer/JsonSerializationVisitor.php line 240 {"exception":"[object] (RuntimeException(code: 0): Your data could not be encoded because it contains invalid UTF8 characters. at /path/to/wallabag/vendor/jms/serializer/src/JMS/Serializer/JsonSerializationVisitor.php:240)"} []
[2018-01-24 16:11:53] request.INFO: Matched route "api_post_entries". {"route":"api_post_entries","route_parameters":{"_controller":"Wallabag\\ApiBundle\\Controller\\EntryRestController::postEntriesAction","_format":"json","_route":"api_post_entries"},"request_uri":"https://domainname/wallabag/web/api/entries.json","method":"POST"} []
[2018-01-24 16:11:53] app.DEBUG: Restricted access config enabled? {"enabled":0} []
```

### Environment

* wallabag version (or git revision) that exhibits the issue: 2.3.1 + 2.3.2
* How did you install wallabag? Via `git clone` or by downloading the package? git clone
* Last wallabag version that did not exhibit the issue (if applicable): -
* php version: 7.0.27
* OS: Debian
* type of hosting (shared or dedicated): shared
* which storage system you choose at install (SQLite, MySQL/MariaDB or PostgreSQL): SQLite

### Steps to reproduce/test case

Add the following link to wallabag: 
https://www.swm.de/dam/swm/pressemitteilungen/2018/01/20180124-swm-mvg-tum-e-trike-fuer-mvg-rad.pdf
Open wallabag.

**Comment by BLeQuerrec** [comment]
Same with that PDF file: http://www.conseil-constitutionnel.fr/conseil-constitutionnel/root/bank/download/2015478QPC2015478qpc_ccc.pdf
In the SQLite database, the file seems to be correctly parsed.

**Comment by Kdecherf** [comment]
Hello,

The two PDF links given as examples are currently leading to 404 errors.

**Comment by eazy2k13** [comment]
The issue seems to be fixed in the recent versions. All PDF documents I've added in the past months are adequatly shown in wallabag and therefore do not trigger any errors or system hangups.
I personally do not have an issue with the provided PDF document link in #4221 .
Thank you for your work.

**Comment by regisb** [comment]
Could it be possible to have a link to the patch(es) that fix this? I am running 2.3.8 and this issue has made the Android app almost unusable for me for the past couple months: I can add new articles but not sync with the server.

I tried to upgrade to the master branch, but failed because of migration errors (see https://github.com/wallabag/wallabag/issues/4169).
